### PR TITLE
Adding support for PaymentMethod

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/model/Charge.java [hunk 1]
```diff
@@ -602,6 +606,473 @@ public void setRuleObject(Rule rule) {
...
+      @SerializedName("swift_code")
+      String swiftCode;
+    }
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class AchDebit extends StripeObject {
+      @SerializedName("account_holder_type")
+      String accountNumber;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Charge.java]: Should be `accountHolderType`?

File: src/main/java/com/stripe/model/Charge.java [hunk 2]
```diff
@@ -602,6 +606,473 @@ public void setRuleObject(Rule rule) {
...
+      String preferredLanguage;
+
+      @SerializedName("verified_name")
+      String verifiedName;
+    }
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class Bitcoin extends StripeObject {
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Charge.java]: Schema is quite different from what autogen has.
```
    public static class Bitcoin extends StripeObject {
      @SerializedName("address")
      String address;

      @SerializedName("amount")
      Long amount;

      @SerializedName("amount_charged")
      Long amountCharged;

      @SerializedName("amount_received")
      Long amountReceived;

      @SerializedName("amount_returned")
      Long amountReturned;

      @SerializedName("refund_address")
      String refundAddress;
    }
```

File: src/main/java/com/stripe/model/Charge.java [hunk 3]
```diff
@@ -602,6 +606,473 @@ public void setRuleObject(Rule rule) {
...
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class Card extends StripeObject {
+      @SerializedName("brand")
+      String brand;
+
+      @SerializedName("checks")
+      PaymentMethod.Card.Checks checks;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Charge.java]: In payflow code, there are actually two classes of checks. One in the details here, and the other one in payment method, eventhough the schemas are exactly the same
```
lib/payment_flows/private/payment_methods/card/checks/details_api_resource.rb
lib/payment_flows/private/payment_methods/card/checks/api_resource.rb 	
```

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Thanks a lot for catching the issues. I have re-generated openapi and added it to the PR so that all tests pass.

Can you do a full review on that one to confirm everything is as expected and ready to merge (beyond a new stripe-mock)?

File: src/main/java/com/stripe/model/PaymentMethod.java [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+        String.format(
+            "%s%s",
+            Stripe.getApiBase(), String.format("/v1/payment_methods/%s/detach", this.getId()));
+    return request(ApiResource.RequestMethod.POST, url, params, PaymentMethod.class, options);
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public class BillingDetails extends StripeObject {
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/PaymentMethod.java]: Missing `static` here for the inner class.

File: src/main/java/com/stripe/model/Charge.java [hunk 5]
```diff
@@ -602,6 +607,487 @@ public void setRuleObject(Rule rule) {
     }
   }
 
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class PaymentMethodDetails extends StripeObject {
+    @SerializedName("account")
+    Account account;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Charge.java]: I think the field name should be `stripe_account` instead?
And we can consider renaming the class below to `StripeAccount`?

**Comment by mickjermsurawong-stripe** [comment]
@remi-stripe 
I did build a jar and does a diff with autogen. The two comments above were results I found.
ptal @remi-stripe

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Thanks for catching those. I fixed the 2 issues, PTAL!

**Comment by mickjermsurawong-stripe** [comment]
@remi-stripe Sorry, I was only looking at `PaymentMethod`, but I think we are missing an expandable field `paymentMethod` field in `PaymentIntent` too? 
The internal site: `docs/api/payment_intents/object#payment_intent_object-payment_method`.
I think we miss that in other typed libs too.

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe so glad you caught this, I just fixed it. PTAL (will fix the other libs)

File: src/main/java/com/stripe/model/Charge.java [hunk 6]
```diff
@@ -30,6 +32,7 @@
   AlternateStatementDescriptors alternateStatementDescriptors;
   @Getter(AccessLevel.NONE) @Setter(AccessLevel.NONE)
       ExpandableField<BalanceTransaction> balanceTransaction;
+  PaymentMethod.BillingDetails billingDetails;
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/Charge.java]: Verified that the backend here does share resource with PaymentMethod

**Comment by mickjermsurawong-stripe** [comment]
thank you @remi-stripe!
LGTM

**Comment by mickjermsurawong-stripe** [comment]
oh I assume you will do version update etc on this PR too?

**Comment by remi-stripe** [comment]
Moved to latest stripe-mock and removed old openapi spec.

**Pull request merged**